service/qos/service_level_controller.hh

```diff
@@ -130,10 +131,11 @@ public:
      * Chack the distributed data for changes in a constant interval and updates
      * the service_levels configuration in accordance (adds, removes, or updates
      * service levels as necessairy).
-     * @param interval - the interval is seconds to check the distributed data.
+     * @param interval_f - lambda function which returns a interval in miliseconds.
+                           The interval is time to check the distributed data.
      * @return a future that is resolved when the update loop stops.
      */
-    void update_from_distributed_data(std::chrono::duration<float> interval);
+    void update_from_distributed_data(std::function<steady_clock_type::duration()> interval_f);
```

Comment:
Why lambda? To support live-update? In that case, why not use updateable_value instead?

Reply:
Well, it does. When you have a specific type for something, it is best to use it, instead of some roughly equivalent generic type. The type used communicates intent. A generic lambda communicates nothing.

Reply:
In principle I don't disagree, it's just that this 10-liner patch has already been in review for a month.

I'm getting the impression that the shorter a patch is, the stricter the reviews.

Reply:
> In principle I don't disagree, it's just that this 10-liner patch has already been in review for a month.

These are rookie numbers. I have a one liner patch, adding an empty line to the docs, waiting for merge for 2+ months.


> I'm getting the impression that the shorter a patch is, the stricter the reviews.

Sometimes it is the case. Fine, im the name of the good-enough principle, I will queue when I get home, but @Jadw1 please fix in a follow-up.